File: Java/프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] ingredient) {
        int count = 0;
        Stack<Integer> stack = new Stack<>();
        
        for (int ing : ingredient) {
            stack.push(ing);
            int j = stack.size();
            if (j >= 4) {
                if (stack.get(j - 4) == 1 &&
                   stack.get(j - 3) == 2 &&
                   stack.get(j - 2) == 3 &&
                   stack.get(j - 1) == 1) {
                    count ++;
                    
                    for (int i = 0; i < 4; i++) stack.pop();
                }
            }
        }
        return count;
    }
}

File: Python3/프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    corr = [1,2,3,1]
    count = 0
    stack = []
    for ing in ingredient:
        stack.append(ing)
        if stack[-4:] == corr:
            count += 1
            for i in range(4):
                stack.pop()
    return count